cmake/IgnSetCompilerFlags.cmake

```diff
@@ -115,11 +115,18 @@ endmacro()
 # Set up compilation flags for GCC or Clang
 macro(ign_setup_gcc_or_clang)
 
+  option(USE_DEFAULT_VISIBILITY_HIDDEN "Hide symbols by default if they are not explicitly specified as visible" FALSE)
+  if(USE_DEFAULT_VISIBILITY_HIDDEN)
+    set(VISIBILITY_FLAG "-fvisibility=hidden")
+  else()
+    set(VISIBILITY_FLAG "-fvisibility")
+  endif()
+
   ign_filter_valid_compiler_options(
     CUSTOM_ALL_FLAGS
         -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal
         -Wshadow -Winit-self -Wswitch-default -Wmissing-include-dirs -pedantic
-        -fvisibility)
+        ${VISIBILITY_FLAG})
```

Comment:
cmake has a visibility property https://cmake.org/cmake/help/v3.16/prop_tgt/LANG_VISIBILITY_PRESET.html, is there a way to make use of that?

Reply:
good call!

should we also set [VISIBILITY_INLINES_HIDDEN](https://cmake.org/cmake/help/v3.16/prop_tgt/VISIBILITY_INLINES_HIDDEN.html#prop_tgt:VISIBILITY_INLINES_HIDDEN)?

Reply:
we are now setting `CMAKE_C_VISIBILITY_PRESET` and `CMAKE_CXX_VISIBILITY_PRESET` as of https://github.com/ignitionrobotics/ign-cmake/pull/196/commits/fd38458199dd12efb2239c7853159a0096f6e55b